lesson01.md

```diff
@@ -1 +1,423 @@
+
+---
+
+### Practice
+
+Let's say we want to go to the cinema with some friends (choose any number).
+
+How many people are going to the cinema in total? Enter it to JavaScript Console.
+<!-- .element: class="fragment" -->
+
+A ticket to watch the movie costs 8€, let JavaScript compute how much we have to pay in total.
+<!-- .element: class="fragment" -->
+
+Harald and Carlo volunteered to pay for the tickets. Use JavaScript to compute how much each has to pay.
+<!-- .element: class="fragment" -->
+
+---
+
+### Data Type: String
+
+* A **String** is a sequence of characters
+* It starts and ends with a `"` double quote
+* **Or** it starts and ends with a `'` single quote
+* Examples: `"Hello"`, `'Sevtap'`
+
+---
+
+### Try it out
+
+Open the console, and type in the following strings:
+
+* Your first name
+* Your favorite food
+* Name of your favorite book / tv series / anime
+
+---
+
+### Strings and Quotes
+
+Strings start and end with  (`'`) or (`"`). But what if we want to add a quote within our string?
+
+```js
+"He said: "Hello""
+```
+
+We must escape the quote with a backslash (`\`).
+
+A backslash in a string means that the character right after the backslash is special:
+
+```js
+"He said: \"Hello\""
+```
+
+---
+
+### Special Characters
+
+If you want a backslash in a string, you need to escape it: `"\\"`
+
+There are some special characters in strings, for example:
+
+* `"\n"` - new line
+* `"\t"` - tab
+
+---
+
+
+### Quiz: Which strings are correct?
+
+```plaintext
+"Hello"
+'World'
+"He said "hello" to me"
+'Let's go!'
+""
+'This is a\nnew line'
+'This is a backslash: \'
+```
+
+---
+
+### Solution
+
+```js
+"Hello"                        // CORRECT
+'World'                        // CORRECT
+"He said "hello" to me";       // WRONG - unescaped quotes
+'Let's go!'                    // WRONG - unescaped '
+""                             // CORRECT
+'This is a\nnew line';         // CORRECT
+'This is a backslash: \';      // WRONG - escaped '
+```
+
+---
+
+### String operators
+
+The addition operator (`+`) can be used to add (concatenate) two strings:
+
+```js
+"a" + "b"    // results: "ab"
+```
+
+---
+
+### String operators
+
+The addition operator can be used on Strings and Numbers:
+
+```js
+1 + 1                // is 2
+"hello" + " world"   // is "hello world"
+```
+
+---
+
+### Importance of data types
+
+```js
+1 + 1        // is 2
+"1" + "1"    // is "11"
+```
+
+Make sure to always choose the correct data type for your data,
+since the operators in JavaScript behave differently depending on the data type.
+
+---
+
+
+### Data Type: Boolean
+
+* **Boolean** describes a yes/no situation
+* in JavaScript, we use `true` or `false`
+
+---
+
+
+### Practice
+
+Open the console, and type in the following:
+
+* `true`
+* `false`
+
+Why would we dedicate an entire data type for just two values?
+
+---
+
+### Comparison operators
+
+For all data types:
+* `===` strict equality
+* `!==` strict inequality
+
+Numbers only:
```

Comment:
Actually the operators below work on strings as well, it will compare alphabetically.
"a" < "b" // true

Reply:
Fixed